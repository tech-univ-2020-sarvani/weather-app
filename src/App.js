import React from 'react';
import './App.css';
import PropTypes from 'prop-types';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import HomePage from './container/HomePage';
import WeatherPage from './container/WeatherPage';
import SearchBar from './container/SearchBar';

function App(props) {
  const { testId } = props;
  return (
    <div className="App" data-testid={testId}>
      <Router>
        <Switch>
          <Route path="/weather">
            <HomePage />
          </Route>
          <Route path="/bycityname">
            <SearchBar />
          </Route>
          <Route path="/">
            <WeatherPage />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}
App.propTypes = {
  testId: PropTypes.string,
};
export default App;
