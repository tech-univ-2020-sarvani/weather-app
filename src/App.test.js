import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('The App', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<App testId="test-app" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
