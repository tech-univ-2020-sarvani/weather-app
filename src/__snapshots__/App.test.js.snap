// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App should render correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-testid="test-app"
  >
    <div
      class="homepage"
    >
      <button
        type="button"
      >
        Predict Weather
      </button>
      <input
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
